modules/ssh/ssh.go

```diff
@@ -145,17 +146,18 @@ func publicKeyHandler(ctx ssh.Context, key ssh.PublicKey) bool {
 		}
 
 		// look for the exact principal
+	principalLoop:
 		for _, principal := range cert.ValidPrincipals {
 			pkey, err := models.SearchPublicKeyByContentExact(principal)
 			if err != nil {
+				if models.IsErrKeyNotExist(err) {
+					log.Debug("Principal Rejected: Unknown Principal: %s", principal)
+					continue principalLoop
```

Comment:
Isn't this will loop forever?

Reply:
No.

https://play.golang.org/p/HA0jVr08_TV